### PR TITLE
[solution]: zh/medium-chunk.md

### PR DESCRIPTION
**Pull request by charleserious**
# Challenge [zh/medium-chunk.md]

## Checklist

- [x] Make sure that you are opening the PR with a single solution (one solution per PR)
- [x] The solution must have at least several references for the readers to dive in depth

### CODE REVIEW

**Comment by ghaiklor** [comment]
Maybe @SUtengda can take a look at this one while @likui628 is absent

File: zh/medium-chunk.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+lang: zh
+level: medium
+tags: tuple
+---
+
+## 挑战
+
+你知道 `loadsh` 吗？
+`Chunk` 是其中一个非常有用的函数，现在我们来实现它。
+`Chunk<T, N>` 接受两个必要类型参数，`T` 必须是一个 `tuple`，`N` 必须是 `integer >= 1`。
```
Comment (SUtengda) [inline zh/medium-chunk.md:13]: Not sure, if `tuple` and `integer >= 1` need to be translated

File: zh/medium-chunk.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+有一个空的累加器，我们将用于一个临时的大块，我们可以开始将 `T` 分割成若干部分。
+这些部分是元组的第一个元素和剩余的部分：
+
+```typescript
+type Chunk<T, N, A extends unknown[] = []> = T extends [infer H, ...infer T]
+  ? never
+  : never
+```
+
+有部分元组 `T`，我们可以检查我们的累加器是否有必要的大小。
```
Comment (SUtengda) [inline zh/medium-chunk.md:48]: Instead of `我们可以检查我们的累加器是否有必要的大小`, I would recommend `我们可以检查累加器的大小是否符合要求`

File: zh/medium-chunk.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+为了达到这个目的，我们在其类型上查询 `length` 属性。
+这很有效，因为我们有一对类型参数 `A` 的能用约束，说它是一个元组。
```
Comment (SUtengda) [inline zh/medium-chunk.md:50]: I would say `这有效，因为我们对类型参数“A”有一个通用约束，它表示一个元组。`

File: zh/medium-chunk.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+```typescript
+type Chunk<T, N, A extends unknown[] = []> = T extends [infer H, ...infer T]
+  ? A['length'] extends N
+  ? never
+  : never
+  : never
+```
+
+如果我们的累加器是空的或没有足够的项目，我们需要继续切分 `T` 直到累加器达到所需要的大小。
```
Comment (SUtengda) [inline zh/medium-chunk.md:60]: `没有足够的项目` I would recommend `没有足够的元素`

**Review by SUtengda — APPROVED** [review]
@ghaiklor I've made some suggestions, but basically it looks good for me

File: zh/medium-chunk.md (hunk 3 above)
Comment (charleserious) [reply zh/medium-chunk.md:50]: Thanks bro~

**Comment by charleserious** [comment]
> @ghaiklor I've made some suggestions, but basically it looks good for me

@SUtengda Thanks bro, great calls~ ❤️

**Review by ghaiklor — APPROVED** [review]

**Comment by ghaiklor** [comment]
Thanks for your contribution and congratulations on the first one! 🎉

**Comment by charleserious** [comment]
> Thanks for your contribution and congratulations on the first one! 🎉

I think you have the wrong person bro~ 😉  @ghaiklor

**Comment by SUtengda** [comment]
> Thanks for your contribution and congratulations on the first one! tada

@charleserious @ghaiklor and thanks for your help, lol

**Pull request merged**